pymodbus/server/simulator/http_server.py

```diff
@@ -300,15 +300,18 @@ async def handle_html(self, request):
 
     async def handle_json(self, request):
         """Handle api registers."""
-        page_type = request.path.split("/")[-1]
-        params = await request.post()
-        json_dict = self.generator_html[page_type][0].copy()
-        result = self.generator_json[page_type][1](params, json_dict)
-        return web.Response(text=f"json build: {page_type} - {params} - {result}")
+        endpoint = request.path.split("/")[-1]
+        params = await request.json()
+        try:
+            result = self.generator_json[endpoint](params)
+        except Exception as exc: # pylint: disable=broad-exception-caught
```

Comment:
We do not like broad exceptions, especially not when it is easy replaced by a strict exception like here.

Reply:
True, and my implementation is against best practices.

However, this one catches unhandled situations. Something like failed int conversion for "range_start/stop", or "value" when calling "Set". The same that aiohttp does by itself, except that it will post the data back in clear-text format, causing problems on Json parsing.

So I have few options:

1. Catch all the known exceptions in the existing code, either here on inplace with or without rethrowing as new, well-defined errors.
2. Leave it as is, handle them as unhandled
3. Combination. Catch all that are known, leave the broad exception in the event of any missed or future implementation errors.

Option 1 and 3 will require me to put in a partial improvement on json side as well, on top of everything.

Please let me know what you think.

Reply:
> on json side

on html side*

Reply:
it is easy to see which exceptions can happen, so please limit it.

Reply:
Changed